nano/node/network.cpp

```diff
@@ -678,9 +700,9 @@ std::deque<std::shared_ptr<nano::transport::channel>> nano::network::list (size_
 }
 
 // Simulating with sqrt_broadcast_simulate shows we only need to broadcast to sqrt(total_peers) random peers in order to successfully publish to everyone with high probability
-std::deque<std::shared_ptr<nano::transport::channel>> nano::network::list_fanout ()
+std::deque<std::shared_ptr<nano::transport::channel>> nano::network::list_fanout (float scale)
 {
-	auto result (list (size_sqrt ()));
+	auto result (list (static_cast<size_t> (std::ceil (scale * size_sqrt ()))));
```

Comment:
`size_sqrt()` returns the ceiling already, maybe it should return a float so that only one rounding up is done?

Reply:
I think I ran in to an issue while testing when the peer count was 1, possibly because sqrt(1.0) over floats yields close, but less than one which would round to zero. Returning a float would work but then all call sites need to cast. If it’s easy though we can do that.

Reply:
There were 2 call sites, one was modified to call `fanout()` directly (see above). Extended test cases in `peer_container.list_fanout` to ensure there are no rounding errors for 1, 2, 1000.